### PR TITLE
Search for plugins in config_base

### PR DESCRIPTION
**Pull request by Ram-Z**
instead of the hardcoded ~/.flexget/plugins. This effectively allows to
put plugins in $XDG_CONFIG_HOME/flexget/plugins.

### CODE REVIEW

File: flexget/plugin.py [hunk 1]
```diff
@@ -356,14 +356,10 @@ def _get_standard_plugins_path():
     :returns: List of directories where plugins should be tried to load from.
     """
     # Get basic path from environment
+    paths = []
     env_path = os.environ.get('FLEXGET_PLUGIN_PATH')
     if env_path:
-        # Get rid of trailing slashes, since Python can't handle them when
-        # it tries to import modules.
```
Comment (gazpachoking) [inline flexget/plugin.py]: Does this comment not apply anymore? Does it break when the env path contains a trailing slash?

File: flexget/plugin.py [hunk 2]
```diff
@@ -416,13 +412,19 @@ def _load_plugins_from_dirs(dirs):
                       'point (before, after). Plugin is not working properly.' % (args[0], phase))
 
 
-def load_plugins():
-    """Load plugins from the standard plugin paths."""
+def load_plugins(extra_dirs=[]):
```
Comment (gazpachoking) [inline flexget/plugin.py]: We can't use a mutable default argument (a list is mutable), as it won't reset if the function is called more than once. Usually, you'd set the default to `None`, then create your list in the body of the function.

File: flexget/plugin.py (hunk 1 above)
Comment (Ram-Z) [inline flexget/plugin.py]: No, I removed the stripping in this function. The trailing slashes were being removed again further down [L377](https://github.com/Ram-Z/Flexget/blob/plugin-path/flexget/plugin.py#L377)

Comment (gazpachoking) [inline flexget/plugin.py]: Gotcha.

**Comment by gazpachoking** [comment]
Cool, thanks! Perhaps we should put a little note about this in upgrade actions since it might affect a few people.

**Pull request merged**